advisor/src/main/kotlin/advisors/Osv.kt

```diff
@@ -0,0 +1,199 @@
+            pkg to request
+        }
+
+        val result = service.getVulnerabilityIdsForPackages(requests.map { it.second })
+        if (result.isFailure) {
+            log.error {
+                "Requesting vulnerabilities IDs for packages failed: ${result.exceptionOrNull()!!.collectMessages()}"
+            }
+        }
+
+        return buildMap {
+            result.getOrNull().orEmpty().forEachIndexed { i, vulnerabilities ->
+                put(requests[i].first.id, vulnerabilities)
+            }
+
+            packages.forEach { pkg ->
+                putIfAbsent(pkg.id, emptyList())
+            }
+        }
+    }
+
+    private fun getVulnerabilitiesForIds(ids: Set<String>): List<Vulnerability> {
+        val result = service.getVulnerabilitiesForIds(ids)
+
+        return result.getOrNull() ?: run {
+            log.error {
+                "Requesting vulnerabilities IDs for packages failed: ${result.exceptionOrNull()!!.collectMessages()}"
+            }
+            emptyList()
+        }
+    }
+}
+
+private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest? {
+    val name: String? = when {
+        pkg.id.namespace.isEmpty() -> pkg.id.name
+        pkg.id.type == "Composer" -> "${pkg.id.namespace}/${pkg.id.name}"
+        else -> "${pkg.id.namespace}:${pkg.id.name}"
+    }
+
+    val ecosystem: String? = when (pkg.id.type) {
+        "Bower" -> null
+        "Composer" -> Ecosystem.PACKAGIST
+        "Crate" -> Ecosystem.CRATES_IO
+        // TODO: The entries for Go should be merged, see https://github.com/oss-review-toolkit/ort/issues/5533.
+        "GoDep" -> Ecosystem.GO
+        "GoMod" -> Ecosystem.GO
+        "NPM" -> Ecosystem.NPM
+        "NuGet" -> Ecosystem.NUGET
+        "Maven" -> Ecosystem.MAVEN
+        "PyPI" -> Ecosystem.PYPI
+        else -> null
+    }
+
+    // Strip the version prefix for GoMod / GoDep packages. That prefix should not be assigned by the package manager
+    // in the first place, see https://github.com/oss-review-toolkit/ort/issues/5532.
+    val version = pkg.id.version.withoutPrefix("v") { pkg.id.version }
+
+    if (!name.isNullOrBlank() && !version.isNullOrBlank() && !ecosystem.isNullOrBlank()) {
+        return VulnerabilitiesForPackageRequest(
+            pkg = org.ossreviewtoolkit.clients.osv.Package(
+                name = name,
+                ecosystem = ecosystem
+            ),
+            version = version
+        )
+    }
+
+    // TODO: Handle C++ projects and / or requesting vulnerabilities by commit-ish.
+
+    return null
+}
+
+private fun Vulnerability.toOrtVulnerability(): org.ossreviewtoolkit.model.Vulnerability {
+    val (scoringSystem, severity) = severity.firstOrNull()?.let {
+        it.type.name to it.score
+    } ?: null to null
+
+    // TODO: Improve parsing the severity properties. Parse vectors, e.g. CVSS:3.1 and consider using the severity
+    // in the database specific property as a fallback.
```

Comment:
FYI: `VulnerabilityReference.getSeverityString()` already has (very) basic parsing for vectors.